internal/cluster/cluster.go

```diff
@@ -703,13 +703,26 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, clusterID strfmt.UU
 		return nil
 	}
 
+	var hostsCount []struct {
+		Count        int
+		Role         models.HostRole
+		Bootstrap    bool
+		CurrentStage models.HostStage
+	}
+	err = m.db.Table("hosts").Select("count(*) as count, role, bootstrap, progress_current_stage as current_stage").
```

Comment:
Inefficient code is not nice.
The problem here is that we bring all the hosts data, in order to use very small portion of it.
If you have better solution, please show it.

Reply:
Ok, my mistake it is not easy to move to join at all. 
We just need to decide if we want to have something generic wrappers or we will need to stop using preload function by default and move to manual query writing.
I think this requires some general code design decision

Reply:
I think it's ugly but better than what we have now.
We should define a minimal host struct (e.g. without the `Inventory` and `FreeAddresses`) and read from DB to that struct instead of the current EagerLoading